WooCommerce/Classes/ViewRelated/Orders/OrdersRootViewController.swift

```diff
@@ -243,6 +310,15 @@ private extension OrdersRootViewController {
         addChild(ordersViewController)
         stackView.addArrangedSubview(contentView)
         ordersViewController.didMove(toParent: self)
+
+        orderListViewModel.$topBanner
+            .sink { [weak self] topBannerType in
+                // Error banner takes preference over the upsell card reader one.
+                if case .error = topBannerType {
+                    self?.hideUpsellCardReaderFeatureAnnouncementView(true)
+                }
+        }
+            .store(in: &cancellables)
```

Comment:
I'm not sure to understand this part, let me know if I'm way off:
- The`$topBanner` property wrapper determines what top banner should be shown by observing changes emitted by the `OrderListViewModel.`
- `.sink` will execute a closure every time the `topBanner` is de-initialized
- `.store(in: &cancellables)` stores when this closure when is de-initialized
- If this de-initialization comes with an error, make the error appear and hide the upsell card reader one.

Reply:
- The$topBanner property wrapper determines what top banner should be shown by observing changes emitted by the OrderListViewModel. ✅ 
- .sink will execute a closure every time the topBanner ~~is de-initialized~~ changes. We are listening for changes in the $topBanner property, so we can react in the case that we get an error. ✅ 
- .store(in: &cancellables) stores when this closure when is de-initialized. We store the cancellable there so we do not lose the closure with which we react to any change in `topBanner` ✅ 
- If this ~~de-initialization~~ change comes with an error, make the error appear and hide the upsell card reader one. ✅

Reply:
Gotcha! Thanks for the clarifications 🙇